src/spec.c

```diff
@@ -1120,206 +1200,380 @@ static void IndexStats_RdbSave(RedisModuleIO *rdb, IndexStats *stats) {
   RedisModule_SaveUnsigned(rdb, stats->termsSize);
 }
 
-// todo: the final solution will scan in background
+#endif // 0
+
+///////////////////////////////////////////////////////////////////////////////////////////////
+
 static threadpool reindexPool = NULL;
 
+typedef struct IndexesScanner {
+  const char *spec_name_opt; // can be null
+  IndexSpec *spec_opt;
+  size_t scannedKeys, totalKeys;
+} IndexesScanner;
+
+static IndexesScanner *IndexesScanner_New(IndexSpec *spec_opt) {
+  IndexesScanner *scanner = rm_calloc(1, sizeof(IndexesScanner));
+  scanner->spec_name_opt = spec_opt ? rm_strdup(spec_opt->name) : NULL;
+  scanner->scannedKeys = 0;
+  RedisModuleCtx *ctx = RedisModule_GetThreadSafeContext(NULL);
+  scanner->totalKeys = RedisModule_DbSize(ctx);
+  RedisModule_FreeThreadSafeContext(ctx);
+  if (spec_opt) {
+    __sync_fetch_and_add(&spec_opt->pending_indexing_ops, 1);
```

Comment:
Why you need it, don't you touch it only with the GIL?

Reply:
This whole complication is because we might have some pending scans (like if we ft.create and immediately ft.alter), so we need to have `tf.info` report we're indexing as long we have pending scans (both global - like rdb-related, or index-specific). I wasn't 100% sure there's no chance we'd not be modifying the counters from different threads simultaneously.